projection/src/main/scala/akka/projection/r2dbc/internal/R2dbcProjectionImpl.scala

```diff
@@ -316,14 +316,16 @@ private[projection] object R2dbcProjectionImpl {
       offsetStore: R2dbcOffsetStore)(implicit
       ec: ExecutionContext,
       system: ActorSystem[_]): FlowWithContext[Envelope, ProjectionContext, Done, ProjectionContext, _] = {
+    val warnAboutFilteredEvents =
+      system.settings.config.getBoolean("akka.projection.r2dbc.warn-about-filtered-events-in-flow")
 
     FlowWithContext[Envelope, ProjectionContext]
       .mapAsync(1) { env =>
         offsetStore
           .isAccepted(env)
           .flatMap { ok =>
             if (ok) {
-              if (skipEnvelope(env)) {
+              if (skipEnvelope(env) && warnAboutFilteredEvents) {
                 log.info("atLeastOnceFlow doesn't support of skipping envelopes. Envelope [{}] still emitted.", env)
```

Comment:
While you are at it, could you change this so that we are not using EventEnvelope.toString. Including persistenceId and seqNr might be enough.

I'm changing toString in https://github.com/akka/akka/pull/31804 but anyway good to not log the full thing here.

Reply:
Tricky because `Envelope` here is a type param with no type bounds at all

Reply:
I see, with the new Akka version it will be ok